### PR TITLE
Bump `multisig-wallet `

### LINKED ISSUE #658: Bump `multisig-wallet`
To:

- `forc 0.42.1`
- `fuel-core 0.18.2`
- `fuels-rs 0.43.0`

### PR DESCRIPTION
**Pull request by K1-R1**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)
- Documentation

## Changes

The following changes have been made:

- Bump application to `forc 0.42.1` and `fuel-core 0.18.2`
- Bump test harness to `fuels 0.43.0`

## Related Issues

Closes #658

### CODE REVIEW

File: multisig-wallet/project/contracts/multisig-contract/src/main.sw [hunk 1]
```diff
@@ -200,6 +200,7 @@ impl Info for Contract {
 /// Recovers a b256 address from each signature;
 /// it then increments the number of approvals by that address' approval weighting.
 /// Returns the final approval count.
+#[inline(never)]
```
Comment (Braqzen) [inline multisig-wallet/project/contracts/multisig-contract/src/main.sw:203]: hm, what did you encounter which made you add this annotation?

**Review by Braqzen — COMMENTED** [review]
Just wondering about the inlining - good to approve after

File: multisig-wallet/project/contracts/multisig-contract/src/main.sw (hunk 1 above)
Comment (K1-R1) [reply multisig-wallet/project/contracts/multisig-contract/src/main.sw:203]: I got a compiler error, which was only resolved with this annotation. After reporting it seems that it was caused by some `asm` in a function called by this function.

Comment (Braqzen) [reply multisig-wallet/project/contracts/multisig-contract/src/main.sw:203]: Sounds like a bug, we should remove the annotation in the future

**Pull request merged**